source/components/NumericInput.js

```diff
@@ -0,0 +1,224 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { isFunction, isString, flow, pick } from 'lodash';
+import Input from './Input';
+import FormField from './FormField';
+
+export default class NumericInput extends FormField {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      caretPosition: 0,
+      separatorsCounter: 0,
+      error: null,
+      oldValue: null,
+    }
+  }
+
+  static SKIN_PARTS = {
+    INPUT: Input.SKIN_PARTS.INPUT,
+  };
+
+  static propTypes = Object.assign({}, FormField.propTypes, {
+    error: PropTypes.string,
+    value: PropTypes.string,
+    placeholder: PropTypes.string,
+    maxLength: PropTypes.number,
+    maxBeforeDot: PropTypes.number,
+    maxAfterDot: PropTypes.number,
+    maxValue: PropTypes.number,
+    rightLabel: PropTypes.string,
```

Comment:
I think we should remove the `rightLabel` feature, as it is very application specific to Daedalus and does not necessarily belong to a general-purpose numeric input. In Daedalus we should create a special component and add the right label feature there.